pandas/core/frame.py

```diff
@@ -528,9 +529,15 @@ def __init__(
                 values = cast_scalar_to_array(
                     (len(index), len(columns)), data, dtype=dtype
                 )
-                mgr = init_ndarray(
-                    values, index, columns, dtype=values.dtype, copy=False
-                )
+                if isinstance(values, list):
+                    # Case 1: values is a list of extension arrays
+                    dtype, _ = infer_dtype_from_array(values[0], pandas_dtype=True)
+                    mgr = arrays_to_mgr(values, columns, index, columns, dtype=dtype)
```

Comment:
Why do you need to get the dtype and pass it to `arrays_to_mgr` here? You already know you have a list of ExtensionArrays (which are properly typed), right? So if you don't pass a dtype, `arrays_to_mgr` should just take the dtype of the passed arrays I think?

Reply:
For some reason I thought that wouldn't work, but it looks like it does. Thanks for the suggestion